doc/whats_new/v0.24.rst

```diff
@@ -26,6 +26,12 @@ random sampling procedures.
   between 32-bits and 64-bits data when the kernel has small positive
   eigenvalues.
 
+- |Fix| In ``fit`` and ``fit_transform`` from
+  :class:`decomposition.TruncatedSVD` and :class:`cross_decomposition.PLSSVD`
+  now initializes the v0 parameter correctly with value sampled from a uniform
+  distribution in `[-1, 1]` to avoid convergence issues with another
+  initialization.
```

Comment:
See my comment below for `PLSSVD`.

For `TruncatedSVD`, the change in this PR does not fix any convergence issues. It fixes the ability or our users to get deterministic results by setting `random_state` parameter and making sure this rng is used to initialize the state of the underlying ARPACK solver.